### PR TITLE
CoilCoolingDX and Related Improvements

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
 - Various fixes/improvements to Coil:Cooling:DX prompted by other related coil work
 - Add a zero flow or zero load check at the top of `CoilCoolingDXCurveFitOperatingMode::CalcOperatingMode`
 - Fix UnitarySystem heating coil sizing with Coil:Cooling:DX:VariableSpeed when system is *not* a heat pump.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by rraustad** [comment]
The first diff shows a gas heating and reheat coil which are now about 1/2 the size of the cooling coil. I have no problem with the change in heating coil size. But what if the cooling coil uses cool reheat, and there is only a large latent load, in this case shouldn't the reheat coil be the same size as the cooling coil?

**Comment by mjwitte** [comment]
The closest thing I can find (UnitarySystem, not a heat pump, autosized, with RH control) is system 1 in UnitarySystem_TwoStageDX.idf
```
 Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 1 FURNACE DX COOL UNITARY SYSTEM, Design Size Nominal Cooling Capacity [W], 6172.67259
 Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 1 FURNACE DX COOL UNITARY SYSTEM, Design Size Nominal Heating Capacity [W], 12875.22776
```
So there the heating coil is nearly 2x the cooling coil capacity. Comparing Sizing:System inputs.
Not really apples to apples 5Zone_Unitary_VSDesuperheatWaterHeater (the one here with the smaller heating coil) is a VAV system with heating flow fraction set to 0.3, thus the smaller coil.
I don't know. This can wait. I'm moving it to v22.2

**Comment by rraustad** [comment]
My point was that it is OK for non-HPs to have different coil sizes, it's just that we need to be sure we size the heating coils appropriately.

**Comment by nrel-bot-2** [comment]
@mjwitte @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mjwitte @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mjwitte @myoldmopar it has been 28 days since this pull request was last updated.

File: src/EnergyPlus/Coils/CoilCoolingDXCurveFitOperatingMode.cc [hunk 1]
```diff
@@ -287,6 +287,18 @@ void CoilCoolingDXCurveFitOperatingMode::CalcOperatingMode(EnergyPlus::EnergyPlu
...
+        outletNode.Temp = inletNode.Temp;
+        outletNode.HumRat = inletNode.HumRat;
+        outletNode.Enthalpy = inletNode.Enthalpy;
+        outletNode.Press = inletNode.Press;
+        OpModeRTF = 0.0;
+        OpModePower = 0.0;
+        OpModeWasteHeat = 0.0;
+        return;
+    }
+
```
Comment (rraustad) [inline src/EnergyPlus/Coils/CoilCoolingDXCurveFitOperatingMode.cc:301]: Already in develop [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/Coils/CoilCoolingDXCurveFitOperatingMode.cc#L294-L303)

File: src/EnergyPlus/UnitarySystem.cc [hunk 2]
```diff
@@ -2170,7 +2170,7 @@ namespace UnitarySystems {
             state.dataSize->DXCoolCap = VariableSpeedCoils::GetCoilCapacityVariableSpeed(
                 state, DataHVACGlobals::cAllCoilTypes(this->m_CoolingCoilType_Num), this->m_CoolingCoilName, ErrFound);
             EqSizing.DesCoolingLoad = state.dataSize->DXCoolCap;
-            EqSizing.DesHeatingLoad = state.dataSize->DXCoolCap;
+            if (this->m_HeatPump) EqSizing.DesHeatingLoad = state.dataSize->DXCoolCap;
```
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.cc:2173]: similar to #9577

**Comment by mjwitte** [comment]
This is moot. Closing.

**Pull request closed without merging**